### PR TITLE
Null checkers and destroy closures

### PR DESCRIPTION
**Pull request by HerlySQR**
Hello, this system is very cool, so I looked more deep into it, and I noticed that in some variables are not checked if they are null (this is more noticeable in lua mode), and the closures are not destroyed, so I also added that.

### CODE REVIEW

File: wurst/AstarJPS.wurst [hunk 1]
```diff
@@ -75,15 +75,17 @@ public class Pathfinder
         return result
 
     function setCond(PathingCondition pred)
-        this.cond = pred
+        if cond != null
```
Comment (Frotty) [inline wurst/AstarJPS.wurst:78]: Is there ever a case or even possibility where one would change the condition mid search?

**Comment by Frotty** [comment]
Thanks. Null checks seem good, however destroying the onFinish closure should perhaps be tied to a parameter, as you could reuse the onFinish closure if it doesn't change, which isn't possible now?

File: wurst/AstarJPS.wurst (hunk 1 above)
Comment (HerlySQR) [reply wurst/AstarJPS.wurst:78]: Not sure, maybe?

**Comment by HerlySQR** [comment]
> Thanks. Null checks seem good, however destroying the onFinish closure should perhaps be tied to a parameter, as you could reuse the onFinish closure if it doesn't change, which isn't possible now?

You're wellcome, I think it should be destroyed because is leaking every time the function search is called, if someone wanna reuse they could use a function and call it inside the closure

**Comment by HerlySQR** [comment]
In these last commits, I'm noticing that also the result path is leaking if is not destroyed, but I didn't include the destroy, because that could affect the maps of creators that uses this system if they saved the path.

By the way, what if in the system the mapmaker can set a limit number of iterations?, to have more control.

**Comment by Frotty** [comment]
> I think it should be destroyed because is leaking every time the function search is called, if someone wanna reuse they could use a function and call it inside the closure

That defeats the point of wanting to reuse the closure, i.e. prevent allocations. I added a parameter and an overriding function, this should serve all demands the best. Thanks for the PR.

**Pull request merged**